bin/yugabyted

```diff
@@ -545,17 +545,29 @@ class ControlScript(object):
                     " Removing...").format(data_dir_files, data_dir))
                 rmcontents(data_dir, exclude_names=[pid_file_name])
 
+            # Add enforced authentication flags on the basis of command-line flag
+            if self.configs.saved_data.get("ysql_enable_auth"):
+                yb_tserver_cmd.extend(["--ysql_enable_auth=true"])
+
+            if self.configs.saved_data.get("use_cassandra_authentication"):
+                yb_tserver_cmd.extend(["--use_cassandra_authentication=true"])
+
             # Add enforced authentication flags on first installation
             # Add ysql_enable_auth, if YSQL_PASSWORD env variable exists
             # Add use_cassandra_authentication, if YCQL_USER or YCQL_PASSWORD env variables exists
             # Add ysql_enable_auth flag to enforce authentication for YSQL
-            if self.setup_env_init.is_exists('YSQL_PASSWORD'):
-                yb_tserver_cmd.extend(["--ysql_enable_auth=true"])
-
-            # Add use_cassandra_authentication flag to enforce authentication for YCQL
-            if self.setup_env_init.is_exists('YCQL_USER') or \
-                    self.setup_env_init.is_exists('YCQL_PASSWORD'):
-                yb_tserver_cmd.extend(["--use_cassandra_authentication=true"])
+            if is_first_install:
+                if self.setup_env_init.is_exists('YSQL_PASSWORD'):
+                    yb_tserver_cmd.extend(["--ysql_enable_auth=true"])
```

Comment:
can we do this as part of the first_install_init

Reply:
also, can we rename first_install_init to first_install_init_auth

Reply:
another possibility is to add this code to the Configs.parse_config_files - when we load config files, we could convert this option to cause an append to tserver_gflags

Reply:
> can we do this as part of the first_install_init

Ahh, that won't be the right approach because we should run this on all nodes but only on the first-run whereas `first_install_init` runs only on the first node and first-run.

Reply:
> also, can we rename first_install_init to first_install_init_auth

Okay. will update the `first_install_init` to `first_install_init_auth`.

Reply:
> another possibility is to add this code to the Configs.parse_config_files - when we load config files, we could convert this option to cause an append to tserver_gflags

I've checked after your suggestion. I guess the best place is `def validate_and_set_configs` (at 1111) to put this snippet because here we're doing verification and overriding the configuration.